src/cursor/cur_table.c

```diff
@@ -837,17 +837,19 @@ __curtable_largest_key(WT_CURSOR *cursor)
 static int
 __curtable_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i;
 
     ctable = (WT_CURSOR_TABLE *)cursor;
     JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
 
-    /* Grab the primary cursor and call bound function. */
-    primary = *ctable->cg_cursors;
```

Comment:
There's a possibility that applying the bounds will fail here, so in this case there's a chance of inconsistent positioning. @luke-pearson What are your thoughts?

Reply:
Call clear on all the cursors that had bounds applied successfully, if clear fails, die.

Reply:
[f5f38c5](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/f5f38c5066ac003c84597c85ca13ca2aef95f23f)